src/_P174_CAN_helper.ino

```diff
@@ -0,0 +1,233 @@
+      Device[deviceCount].SendDataOption     = true;
+      Device[deviceCount].TimerOption        = false;
+      Device[deviceCount].GlobalSyncOption   = false;
+      Device[deviceCount].TimerOptional      = true;
+      Device[deviceCount].Custom             = true;
+      Device[deviceCount].DecimalsOnly       = true;
+      break;
+    }
+
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_174);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      const uint8_t valueCount = getValueCountFromSensorType(static_cast<Sensor_VType>(PCONFIG(1)));
+      if (valueCount > 0) {
+        String lines[valueCount];
+        LoadCustomTaskSettings(event->TaskIndex, lines, valueCount, 0);
+
+        for (int i = 0; i < valueCount; ++i) {
+        if (lines[i].length() == 0) {
+          lines[i] = concat(F("val"), i + 1);
+        }
+          ExtraTaskSettings.setTaskDeviceValueName(i, lines[i]);
+        }
+      }
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUECOUNT:
+    {
+      event->Par1 = getValueCountFromSensorType(static_cast<Sensor_VType>(PCONFIG(1)));
+      success = true;
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVTYPE:
+    {
+      event->sensorType = static_cast<Sensor_VType>(PCONFIG(1));
+      event->idx        = 1;
+      success           = true;
+      break;
+    }
+
+    case PLUGIN_SET_DEFAULTS:
+    {
+      PCONFIG(0) = 0;
+      PCONFIG(1) = static_cast<int16_t>(Sensor_VType::SENSOR_TYPE_NONE);
+      PCONFIG(2) = 0;
+      break;
+    }
+
+    case PLUGIN_FORMAT_USERVAR:
+    {
+      string = UserVar.getAsString(event->TaskIndex, event->idx, static_cast<Sensor_VType>(PCONFIG(1)), 2);
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      String lines[VARS_PER_TASK];
+
+      addRowLabel(F("Node"));
+      addNumericBox(F("p174_node"), PCONFIG(0), 0, 0xFFFF);
+
+      addRowLabel(F("Task Id"));
+      addNumericBox(F("p174_task_id"), PCONFIG(2) + 1, 0, 0xFFFF);
+
+      addRowLabel(F("Sensor Type"));
+      addTextBox(F("p174_sensor_type"),
+                getSensorTypeLabel(static_cast<Sensor_VType>(PCONFIG(1))),
+                32,
+                true);
+
+      LoadCustomTaskSettings(event->TaskIndex, lines, VARS_PER_TASK, 0);
+
+      for (int i = 0; i < VARS_PER_TASK; ++i) {
+        addRowLabel(concat(F("Label "), i + 1));
+        if (lines[i].length() == 0) {
+          lines[i] = concat(F("val"), i + 1);
+        }
+        addTextBox(concat(F("p174_value"), i), lines[i], 0);
+      }
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      String lines[VARS_PER_TASK];
+
+      addRowLabel(F("Node"));
+      PCONFIG(0) = getFormItemInt(F("p174_node"));
+      PCONFIG(2) = getFormItemInt(F("p174_task_id"))-1;
+
+      for (int i = 0; i < VARS_PER_TASK; ++i) {
+        lines[i] = webArg(concat(F("p174_value"), i));
+        if (lines[i].length() == 0) {
```

Comment:
The String class has an `isEmpty()` method available (multiple occurrences found)